--- a/web/src/Controller/Authentication.php
+++ b/web/src/Controller/Authentication.php
@@ -34,9 +34,14 @@
         $success = false;
         $template_vars = [];
 
-        if ($request->isMethod('POST')) {
-            $user = $request->request->get('user');
-            $password = $request->request->get('password');
+        if ($request->isMethod('POST') || isset($_SERVER['PHP_AUTH_USER'])) {
+            if (isset($_SERVER['PHP_AUTH_USER']) && isset($_SERVER['PHP_AUTH_PW'])) {
+                $user = $_SERVER['PHP_AUTH_USER'];
+                $password = $_SERVER['PHP_AUTH_PW'];
+            } else {
+                $user = $request->request->get('user');
+                $password = $request->request->get('password');
+            }
 
             if (empty($user) || empty($password)) {
                 $template_vars['error'] = $app['translator']->trans('messages.error_empty_fields');
