--- a/web/app/controllers.php
+++ b/web/app/controllers.php
@@ -58,14 +58,20 @@
     // processing the request
     if ($app['session']->has('username')) {
         $username = $app['session']->get('username');
-        $preferences = $app['preferences.repository']->userPreferences($username);
-        $app['user.preferences'] = $preferences;
-        $app['user.timezone'] = $preferences->get('timezone');
-
-        // Set application language
-        $request->setLocale($preferences->get('language'));
-        $app['translator']->setLocale($preferences->get('language'));
-        return;
+        // Clear user session if HTTP authentication changed
+        if (isset($_SERVER['PHP_AUTH_USER'])
+        && $username != $_SERVER['PHP_AUTH_USER']) {
+            $app['session']->clear();
+         } else {
+                $preferences = $app['preferences.repository']->userPreferences($username);
+                $app['user.preferences'] = $preferences;
+                $app['user.timezone'] = $preferences->get('timezone');
+
+            // Set application language
+            $request->setLocale($preferences->get('language'));
+            $app['translator']->setLocale($preferences->get('language'));
+            return;
+        }
     }
 
     if (isset($app['auth.methods'])) {
